server/command/command.go

```diff
@@ -0,0 +1,63 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/pluginapi"
+)
+
+type Handler struct {
+	client *pluginapi.Client
+}
+
+type Command interface {
+	Handle(args *model.CommandArgs) (*model.CommandResponse, *model.AppError)
+	executeHelloCommand(args *model.CommandArgs) *model.CommandResponse
+}
+
+const helloCommandTrigger = "hello"
+
+// Register all your slash commands in the NewCommandHandler function.
+func NewCommandHandler(client *pluginapi.Client) Command {
+	err := client.SlashCommand.Register(&model.Command{
+		Trigger:          helloCommandTrigger,
+		AutoComplete:     true,
+		AutoCompleteDesc: "Say hello to someone",
+		AutoCompleteHint: "<username>",
```

Comment:
I wonder if we should illustrate the advanced autocomplete functionality?

Reply:
What is that advanced functionality? I'm not aware, is there something I can read?

Reply:
Take a peek at https://github.com/mattermost/mattermost-plugin-playbooks/blob/6c50600fb6151ea1abfed82961e980131b0ca2ab/server/command/command.go#L61 (ironically in an unnecessary `command` package itself ;P)